artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/AbstractSequentialFileFactory.java

```diff
@@ -224,8 +224,10 @@ public void releaseBuffer(final ByteBuffer buffer) {
    @Override
    public void createDirs() throws Exception {
       boolean ok = journalDir.mkdirs();
-      if (!ok) {
-         throw new IOException("Failed to create directory " + journalDir);
+      if (!ok && !journalDir.exists()) {
```

Comment:
shouldn't this be ||

if !ok.. and directory exists.. shouldn't this throw an error?

and how home mkdirs worked and the directory did not exist?

Reply:
@jbertram Looks good.  I would consider also wrapping the whole thing in a try, catch(Throwable).  Different JVM implementations might handle the IO Exception differently e.g. throw a RuntimeException.